partner_firstname/tests/test_partner_firstname.py

```diff
@@ -25,129 +24,73 @@
-            'Copy of the user failed with wrong name'
-        )
-        self.assertEqual(
-            vals['lastname'], _('%s (copy)') %
-            'lastname', 'Copy of the user failed with wrong lastname')
-        self.assertEqual(vals['firstname'], 'firstname',
-                         'Copy of the user failed with wrong firstname')
-
-    def test_update_user_lastname(self):
-        cr, uid, context = self.cr, self.uid, self.context
-        # create a user
-        res_id = self.user_model.create(
-            cr, uid, self.fields_user, context=context)
-        # get the related partner id and add it a firstname
-        flds = self.user_model.read(
-            cr, uid, [res_id], ['partner_id'], context=context)[0]
-        self.partner_model.write(
-            cr, uid, flds['partner_id'][0], {'firstname': 'firstname'},
-            context=context)
-        self.user_model.write(
-            cr, uid, res_id, {'name': 'change firstname'}, context=context)
-        vals = self.user_model.read(
-            cr, uid, [res_id], ['name', 'lastname', 'firstname'],
-            context=context)[0]
-
-        self.assertEqual(vals['name'], 'change firstname',
-                         'Update of the user lastname failed with wrong name')
-        self.assertEqual(
-            vals['lastname'], 'change',
-            'Update of the user lastname failed with wrong lastname')
-        self.assertEqual(
-            vals['firstname'], 'firstname',
-            'Update of the user lastname failed with wrong firstname')
-
-    def test_update_user_firstname(self):
-        cr, uid, context = self.cr, self.uid, self.context
-        # create a user
-        res_id = self.user_model.create(
-            cr, uid, self.fields_user, context=context)
-        # get the related partner id and add it a firstname
-        flds = self.user_model.read(
-            cr, uid, [res_id], ['partner_id'], context=context)[0]
-        self.partner_model.write(
-            cr, uid, flds['partner_id'][0], {'firstname': 'firstname'},
-            context=context)
-        self.user_model.write(
-            cr, uid, res_id, {'name': 'lastname other'}, context=context)
-        vals = self.user_model.read(
-            cr, uid, [res_id], ['name', 'lastname', 'firstname'],
-            context=context)[0]
-
-        self.assertEqual(
-            vals['name'], 'lastname other',
-            'Update of the user firstname failed with wrong name')
-        self.assertEqual(
-            vals['lastname'], 'lastname other',
-            'Update of the user firstname failed with wrong lastname')
-        self.assertFalse(
-            vals['firstname'],
-            'Update of the user firstname failed with wrong firstname')
+        super(PartnerFirstnameCase, self).setUp()
+        self.check_fields = True
+        self.create_original()
+
+    def create_original(self):
+        self.original = self.env["res.partner"].create({
+            "lastname": "lastname",
```

Comment:
I would strongly advise to use some unicode string with special char in tests case. I should prevent infamous encoding error regressions.